app/views/users/show.html.erb

```diff
@@ -1,2 +1,34 @@
-<h1>User Profile</h1>
-<!-- Placeholder for displaying single user profile -->
\ No newline at end of file
+<main>
+  <div class="user-container">
+    <div class="profile-pic">
+      <img src="<%= @user.photo %>" alt="profile-picture">
+    </div>
+    <div class="user-info">
+      <h2><%= @user.name %></h2>
+      <p class="posts-counter">Number of posts: <%= @user.posts_counter || 0 %></p>
+    </div>
+  </div>
+  <article class="user-bio">
+    <h5>Bio</h5>
+    <div class="bio"><%= @user.bio %></div>
+  </article>
+  <article>
+    <% @posts.each do |post| %>
+      <%= link_to post.title, user_post_path(@user, post) %>
+      <div class="post-container">
+        <h5 class="post-title"><%= post.title %></h5>
+        <% if post.text.present? %>
+          <p><%= @post.text[0..100] %>...</p>
+        <% else %>
+         <p>No text available...</p>
+        <% end %>
+      </div>
+      <div class="comment-likes-div">
+        <p>Comment: <%= post.comments_counter || 0 %>, Likes: <%= post.likes_counter || 0 %></p>
+      </div>
+    <% end %>
+  </article>
+  <div>
+    <%= link_to "See all posts", user_posts_path(@user) %>
+  </div>
+</main>
```

Comment:
- Since no user is displayed in the index page, it won't be able to find a single user. Please, kindly fix this too.